stdlib/ballerina-builtin/src/main/ballerina/ballerina/io/natives.bal

```diff
@@ -75,85 +94,64 @@ public native function createCharacterChannel (ByteChannel byteChannel, string e
-@Param {value:"content: Text content which should be written"}
-@Param {value:"startOffset: If the content needs to be written with an offset, the value of that offset"}
-@Return {value:"Number of characters written"}
-public native function <CharacterChannel channel> writeCharacters (string content, int startOffset) (int);
-
-@Description {value:"Function to close a character channel"}
-@Param {value:"channel: The CharacterChannel to be closed"}
-public native function <CharacterChannel channel> closeCharacterChannel ();
+public native function createDelimitedRecordChannel (CharacterChannel channel,
+                                                     string recordSeparator,
+                                                     string fieldSeparator) (DelimitedRecordChannel);
 
 @Description {value:"Function to read bytes"}
 @Param {value:"channel: The ByteChannel to read bytes from"}
 @Param {value:"numberOfBytes: Number of bytes which should be read"}
 @Return {value:"The bytes which were read"}
 @Return {value:"Number of bytes read"}
-public native function <ByteChannel channel> readBytes (int numberOfBytes) (blob, int);
-
-@Description {value:"Function to read all bytes in the given I/O source"}
-@Return {value:"all bytes read from the channel"}
-@Return {value:"Number of bytes read"}
-public native function <ByteChannel channel> readAllBytes () (blob, int);
+@Return {value:"Returns if there's any error while performaing I/O operation"}
+public native function <ByteChannel channel> read (int numberOfBytes, int offset) (blob, int, IOError);
 
 @Description {value:"Function to write bytes"}
 @Param {value:"channel: The ByteChannel to write bytes to"}
 @Param {value:"content: Bytes which should be written"}
 @Param {value:"startOffset: If the bytes need to be written with an offset, the value of that offset"}
 @Return {value:"Number of bytes written"}
-public native function <ByteChannel channel> writeBytes (blob content, int startOffset) (int);
+public native function <ByteChannel channel> write (blob content, int startOffset, int numberOfBytes) (int, IOError);
 
-@Description {value:"Function to close a byte channel"}
-@Param {value:"channel: The ByteChannel to be closed"}
-public native function <ByteChannel channel> close ();
+@Description {value:"Function to read characters"}
+@Param {value:"channel: The CharacterChannel to read characters from"}
+@Param {value:"numberOfChars: Number of characters which should be read"}
+@Return {value:"The character sequence which was read"}
+public native function <CharacterChannel channel> readCharacters (int numberOfChars) (string, IOError);
+
+@Description {value:"Function to write characters"}
+@Param {value:"channel: The CharacterChannel which will allow writing characters"}
+@Param {value:"content: Text content which should be written"}
+@Param {value:"startOffset: If the content needs to be written with an offset, the value of that offset"}
+@Return {value:"Number of characters written"}
```

Comment:
Document the returned error as well

Reply:
Added the return with 3b58422f6c457c12c1ed16661b17ae49a52a17a7